### PR TITLE
expression: add new scalar function IsTruthWithNull (#19621)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #19621 to release-3.0

---

### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/17720, close https://github.com/pingcap/tidb/issues/17476 

Problem Summary: NewFunction can't get keepnull attributes for isture function.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed: 
1. Add a new scalar function name IsTruthWithNull in parser.
2. Use the function name to build ScalarFuncton in wrapWithIsTrue.

How it Works:

### Related changes

- Need to cherry-pick to the release branch 3.0,4.0

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- Fix the behavior of rewrite ScalarFunction IsTure .

### CODE REVIEW

File: expression/builtin.go [hunk 1]
```diff
@@ -607,4 +608,45 @@ var funcs = map[string]functionClass{
 	ast.TiDBIsDDLOwner: &tidbIsDDLOwnerFunctionClass{baseFunctionClass{ast.TiDBIsDDLOwner, 0, 0}},
 	ast.TiDBParseTso:   &tidbParseTsoFunctionClass{baseFunctionClass{ast.TiDBParseTso, 1, 1}},
 	ast.TiDBDecodePlan: &tidbDecodePlanFunctionClass{baseFunctionClass{ast.TiDBDecodePlan, 1, 1}},
+<<<<<<< HEAD
```
Comment (qw4990) [inline expression/builtin.go:611]: @wshwsh12 Please fix it.

**Review by XuHuaiyu — DISMISSED** [review]
LGTM

**Review by SunRunAway — DISMISSED** [review]
LGTM

**Review by bb7133 — APPROVED** [review]

**Pull request merged**